### PR TITLE
enforce authenticated request bodies for Admin-API

### PR DESCRIPTION
**Pull request by aead**
## Description
This commit adds a check to the server's admin-API such that it only
accepts Admin-API requests with authenticated bodies. Further this
commit updates the `madmin` package to always add the
`X-Amz-Content-Sha256` header.

This change improves the Admin-API security since the server does not
accept unauthenticated request bodies anymore.

After this commit `mc` must be updated to the new `madmin` api because
requests over TLS connections will fail.

## Motivation and Context
Security

## How Has This Been Tested?
manually

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

File: docs/bucket/notifications/README.md [hunk 1]
```diff
@@ -99,7 +99,7 @@ connection = pika.BlockingConnection(pika.ConnectionParameters(
 channel = connection.channel()
 
 channel.exchange_declare(exchange='bucketevents',
-                         type='fanout')
+                         exchange_type='fanout')
```
Comment (krisis) [inline docs/bucket/notifications/README.md]: this seems unrelated. can we do this in a different PR?

**Review by krisis — COMMENTED** [review]
Overall changes LGTM. We could do the bucket notifications change in a different PR.

**Review by donatello — DISMISSED** [review]
LGTM - but same nit as @krisis - unrelated changes are better in a different PR.

**Comment by aead** [comment]
Huh, how did 631ec2b7320ed4670faa5a26a6633e77c027d620 managed to get into this PR?! I usually pull from master and `co -b`... Will remove of course, thanks @donatello @krisis

**Review by krisis — APPROVED** [review]

**Review by donatello — APPROVED** [review]
